astropy/utils/masked/function_helpers.py

```diff
@@ -133,10 +133,18 @@
 }  # fmt: skip
 
 # Really should do these...
-IGNORED_FUNCTIONS |= {
-    getattr(np, setopsname) for setopsname in np.lib.arraysetops.__all__
-}
-
+if NUMPY_LT_2_0:
+    NP_LIB_ARRAYSETOP_FUNCS = np.lib.arraysetops.__all__
+else:
+    # Public set operations have been moved to the top-level namespace in numpy 2.0
```

Comment:
I'd prefer not to list them explicitly, but rather get them from the implementation. That way, if another function is added to their `__all__`, we'll notice (or in this case, know to continue to ignore it...).

I'm happy to revisit this after numpy 2.0 has actually appeared...

Reply:
@neutrinoceros , are you able to make the requested change? Thank you! 🙏

Reply:
Actually we'd not notice as any new functions would automatically be added to `IGNORED_FUNCTIONS` while otherwise the CI would probably fail, pointing us to add it manually; but I think this way it's just as well.